### ISSUE TITLE
Filtering calendar export

### ISSUE THREAD

**Issue by jumper047** [mpcjanssen/simpletask-android#461]
Please, add ability to automatically set reminders only for tasks with specific attributes.

**Comment by vojtechkral** [comment]
This could be done by leveraging existing filter infrastracture ... ie. create a named fileter and tell calendar sync code to use that for filtering. However I'm going to have to get the calendar code in shape in the first place...

**Comment by mpcjanssen** [comment]
There is already a specific attribute for exporting a calendar item (due: date) it's also possible to make reminders for single tasks. @jumper047 please provide an example of a workflow or a usecase where this is not enough?

**Comment by jumper047** [comment]
Let's say I have entries in the inbox - it is necessary to buy a new ear pads for headphones, for documents to come on Friday and owncloud update on the server. After processing, they are transformed:
(D) to buy a new ear pads for headphones @money t: 2016-07-11
(B) on the server to update owncloud
(A) to come for documents t: 2016-07-03

The point is that the time in the problem (D) - a time when I have more time to think about it. If successful scenario I have money, I climbed into @money tag and see her.
Time in the problem (A) - the time when I need to accomplish the task.
So, i need calendar appointment fo third and dont need for first.

**Comment by vojtechkral** [comment]
@jumper047 BTW, you are using calendar reminders for tasks with a threshold date?

**Comment by jumper047** [comment]
Yes. Is this wrong?

13 июня 2016 г. 9:37:58 GMT+03:00, Vojtech Kral <notifications@github.com> пишет:
>@jumper047 BTW, you are using calendar reminders for tasks with a
>threshold date?
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/mpcjanssen/simpletask-android/issues/461#issuecomment-225499790

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by vojtechkral** [comment]
@jumper047 No, it's now wrong, it's just that I previously thought no one was using threshold sync and planned to remove it. I guess I'll preserve it then...

**Comment by smichel17** [comment]
@jumper047 Out of curiosity, why use the reminder for threshold date instead of due date?

**Comment by jumper047** [comment]
I always thought, that threshold is date I planned to do something, and due is deadline. Is it wrong? Really,I never use due. Seems like I am doing something wrong :/.

16 июня 2016 г. 20:09:35 GMT+03:00, smichel17 <notifications@github.com> пишет:
>@jumper047 Out of curiosity, why use the reminder for threshold date
>instead of due date?
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/mpcjanssen/simpletask-android/issues/461#issuecomment-226550336

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by ddisanti** [comment]
@jumper047 I don't think you're doing something wrong. SimpleTask and the todo.txt "standard" are so flexible that allow they people to have very different approaches or workflows. And you should use the one that suits you better.

**Comment by smichel17** [comment]
+1 to what @ddisanti said. There is no right or wrong. That said, if you never use due date, @vojtechkral could remove threshold reminders and you could just use due date as a replacement. Here's how I use threshold and due dates:

I have always thought of threshold date as "I can't do this task or don't need to think about this task until then."

- For example, I have a task related to upgrading my operating system to Fedora 24 (from Fedora 23).
  - Fedora 24 will not be released until June 21st, so that's not a task I can do right now, so I don't really need to think about it right now.
  - That task has a threshold date of June 21st, and my filters do not show tasks with threshold dates in the future, so it will stay hidden until then.

I use due date as a deadline. If I decide I want to do a task on a certain date, that's still a deadline -- it's just one I set for myself. 

So, every task that is past its threshold date and before its due date is one that I am able to work on. Tasks with a deadline of today take priority because they either need to be completed that day or I decided I would complete them on that day (although I do that less often; I use GTD, so I make decisions about which long-term projects to work on daily, unless there's an external commitment that requires me to plan in advance).

**Comment by mpcjanssen** [comment]
As @ddisanti I use threshold dates in exactly the same way as @smichel17 . I almost never use due dates, time critical tasks are in my calendar.

**Comment by smichel17** [comment]
I'll also mention, I also use threshold dates like GTD's tickler file for someday/maybe projects and tasks. Things that I'm definitely not going to do right now but I want to be reminded of in a couple weeks or months. For example, a language I want to learn. 

I have a filter set up called 'INBOX' that shows all tasks with no lists or no tags (it's actually more complicated but that's the gist of it). I review this daily. So if I set something with a threshold date in the future and without a list or without a tag, it'll appear in my inbox on that day.

**Comment by jumper047** [comment]
Ok, i understand:). But the main reason for me to use thershold as planned
date - ability to hide tasks in future. I have filter named "today", that
includes tasks without date, and tasks with current date. If i planned
something for today week ago, i'll 1) got it in calendar 2)automatically
have it in "today"list, when it needed. In simpletask, i can't to do it
with due.

2016-06-17 20:47 GMT+03:00 smichel17 <notifications@github.com>:

**Comment by jumper047** [comment]
Hmm, by the way - emacs org-mode has nice feaure - block agenda.  Some
blocks with different filters in one list. May be it is possible to add
something like this to SimpleTask?

2016-06-18 2:51 GMT+03:00 Dmitriy Pshonko <jumper047@gmail.com>:

**Comment by vojtechkral** [comment]
Okay, I'll see if I could keep threshold sync as well...

**Comment by mpcjanssen** [comment]
@vojtechkral it should be easy to retrieve a date from a task based on due date and threshold date in such a way that you always get the right one for putting it on the calendar (for example due date wins)
Add an extension method getCalendarDate to Task in the CalendarSync.kt use that and threshold and due date differences are gone and the functionallity is easily extensible to other future dates too.

**Comment by mpcjanssen** [comment]
@vojtechkral something like:

```
fun Task.getCalendarDate(sync_type: Int) : DateTime? {
    return this.dueDate?.toDateTime()

}
```

**Comment by smichel17** [comment]
@jumper047 I believe you can actually do that with due, using a Lua script.

**Comment by jumper047** [comment]
@smichel17 , yes, i can:). So, my case is not necessary now - i will use due date as sheduled, and threshold as "think later about it".

**Comment by mpcjanssen** [comment]
@jumper047 this will be possible to implement yourself after the `onCalendarSync` of  #349 is implemented.

**Comment by mpcjanssen** [comment]
Closed, this will be possible using Lua (#349) and until then not at all.
